doc/reference/api/overview.rst

```diff
@@ -232,7 +232,7 @@ current :ref:`stability level <api_lifecycle>`.
      - 1.0
      - 2.6
 
-   * - :ref:`pinctrl_api`
+   * - :ref:`pinctrl-guide`
```

Comment:
This feels a bit premature. I'm not against a better guide at all, but we generally link only within doc/reference from this table. Why not just use `:no-link:` from doc/guides/pinctrl/index.rst, and keep the reference page around instead?

Reply:
We can't have duplicate C definitions, so we either inline or keep them in reference.

Reply:
That is exactly what `:no-link:` seems to be meant to avoid:

``` 
no-link

    Instructs Breathe to not attempt to generate any document targets for the content generated by this particular directive.

    This allows you to have your main reference listings somewhere with targets, but then to be able to sneak in repeat directives into other parts of the documentation to illustrate particular points without Sphinx getting confused what should be linked to by other references.
```

But I tried it and it doesn't seem to work the way I expected.

That said, I still think it's weird to have this exception where everything else is under the API reference, except for pinctrl. I get what you're trying to do here, but I think this shouldn't be an exception to the pattern. Can we try to figure out a better way to organize things as part of the larger docs rework?